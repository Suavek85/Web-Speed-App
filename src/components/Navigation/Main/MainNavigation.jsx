import React from 'react'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import { desktopWidth } from'../../../scss/mediaqueries'
import { Link } from 'react-router-dom'
import MenuLink from '../../../utils/MenuLink/MenuLink'
import NavOpenLink from '../../Navigation/Side/NavOpenLink/NavOpenLink'
import ToggleSwitch from '../../ToggleSwitch/ToggleSwitch'
import { APPTITLE, SIGN_IN } from '../../../constants/content'
import { smallMobileWidth, mobileWidth } from '../../../scss/mediaqueries'
import styles from './MainNavigation.scss'

export default function MainNavgation() {

  const isDesktop = useMediaQuery(desktopWidth)
  const settingsLink = <h2>Settings</h2>
  const settingsIcon = (
    <>
      <div className={ styles.blockMenu }></div>
      <div className={ styles.blockMenu }></div>
      <div className={ styles.blockMenu }></div>
    </>
  )

  //STYLE
  const cx = classNames.bind(styles)
  
  const getClass = cx({
    block: true,
    blockMobile: useMediaQuery(mobileWidth),
    blockSmallMobile: useMediaQuery(smallMobileWidth),
  })
  
  //if mobile/tablet inside SideNavigation render SignInLink (create a component for that) and toggleSwitch

  return (
    <section className={ getClass } >
      <Link to="/" className={ styles.blockTitle }>
        <h1>{ APPTITLE }</h1>
      </Link>
      <nav className={ styles.blockLinks }>
        {isDesktop && (
          <>
            <MenuLink linkPath="/signin" content={ SIGN_IN } />
            <NavOpenLink settings={settingsLink} />
            <ToggleSwitch />
          </>
        )}
        {!isDesktop && (
          <NavOpenLink settings={settingsIcon} />
        )}
      </nav> 
    </section>
  )
}