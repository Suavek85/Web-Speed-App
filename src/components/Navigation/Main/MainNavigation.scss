@import '../../../scss/styles';

$block: mainnav;
$blockTitle: elem($block, title);
$blockLinks: elem($block, links);
$blockMenu: elem($block, menu);
$blockMobile: modifier($block, mobile);
$blockSmallMobile: modifier($block, small-mobile);

.#{$block} {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-right: 20px;
  padding-left: 20px;
  margin-bottom: 20px;
  background-color: var(--color-main-background);

  h1 {
    font-weight: normal;
    color: var(--color-header-font);
  }

  h2 {
    padding: 0 5px;
    font-size: em(18);
    font-weight: normal;
    color: var(--color-header-font);
  }

  h2:hover {
    color: var(--color-header-focus-font);
  }

  &--mobile {
    h1 {
      font-size: em(19);
    }

    h2 {
      font-size: em(16);
    }
  }

  &--small-mobile {
    padding-right: 8px;
    padding-left: 8px;

    h1 {
      font-size: em(18);
    }

    h2 {
      font-size: em(15);
    }
  }

  &__title {
    text-decoration: none;
  }

  &__links {
    display: flex;
  }

  &__menu {
    width: 35px;
    height: 4px;
    margin: 6px 0;
    background-color: var(--color-header-font);
  }
}

:export {
  block: $block;
  blockTitle: $blockTitle;
  blockLinks: $blockLinks;
  blockMenu: $blockMenu;
  blockMobile: $blockMobile;
  blockSmallMobile: $blockSmallMobile;
}
