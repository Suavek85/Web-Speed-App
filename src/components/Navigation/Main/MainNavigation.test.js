import React from 'react'
import { shallow } from 'enzyme'
import MainNavigation from './MainNavigation'

it('Should match the snapshot', () => {
  const wrapper = shallow(<MainNavigation />)
  expect(wrapper).toMatchSnapshot()
})