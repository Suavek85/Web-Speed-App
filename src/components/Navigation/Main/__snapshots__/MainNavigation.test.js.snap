// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<section
  className="block"
>
  <Link
    to="/"
  >
    <h1>
      Web Speed App
    </h1>
  </Link>
  <nav>
    <NavOpenLink
      settings={
        <React.Fragment>
          <div />
          <div />
          <div />
        </React.Fragment>
      }
    />
  </nav>
</section>
`;
